src/third_party/cudasift/CMakeLists.txt

```diff
@@ -0,0 +1,75 @@
+cmake_minimum_required(VERSION 2.8)
+
+project(cudasift)
+set(cudaSift_VERSION_MAJOR 2)
+set(cudaSift_VERSION_MINOR 0)
+set(cudaSift_VERSION_PATCH 0)
+
+set(CPACK_PACKAGE_VERSION_MAJOR "${cudaSift_VERSION_MAJOR}")
+set(CPACK_PACKAGE_VERSION_MINOR "${cudaSift_VERSION_MINOR}")
+set(CPACK_PACKAGE_VERSION_PATCH "${cudaSift_VERSION_PATCH}")
+set(CPACK_GENERATOR "ZIP")
+include(CPack)
+
+if (WIN32)
+  set(EXTRA_CXX_FLAGS "/DVERBOSE /D_CRT_SECURE_NO_WARNINGS ")
+  list(APPEND CUDA_NVCC_FLAGS "-arch=sm_35;--compiler-options;-O2;-DVERBOSE") 
```

Comment:
I would also suggest to extract the GPU architecture to a separate variable or automate its detection if possible. As it is definitely not backward compatible, i would also question its forward compatibility, as i spent a good amount of time trying to debug an issue of the sort in mdaiter's fork: https://github.com/mdaiter/openMVG/issues/12